tests/src/app-promotion.test.ts

```diff
@@ -0,0 +1,375 @@
+  });
+
+  it('for different accounts in one block is possible', async () => {
+    // arrange: Alice, Bob, Charlie, Dave balance = 1000
+    // arrange: Alice, Bob, Charlie, Dave stakes 100
+
+    // act:     Alice, Bob, Charlie, Dave calls appPromotion.unstake(100)
+
+    // assert:  query appPromotion.pendingUnstake(Alice/Bob,Charlie,Dave) to equal []
+  });
+});
+
+describe.skip('Admin adress', () => {
+  it('can be set by sudo only', async () => {
+    // assert:  Sudo calls appPromotion.setAdminAddress(Alice) /// Sudo successfully sets Alice as admin
+    // assert:  Bob calls appPromotion.setAdminAddress(Bob) throws /// Random account can not set admin
+    // assert:  Alice calls appPromotion.setAdminAddress(Bob) throws /// Admin account can not set admin
+  });
+
+  it('can be Substrate address only', async () => {
+    // assert:  Sudo calls appPromotion.setAdminAddress(0x0...) throws
+  });
+
+  it('can be unset by sudo', async () => {
+    // TODO can he? What if admin address compromised?
+  });
+});
+
+describe.skip('App-promotion collection sponsoring', () => {
+  it('can not be set by non admin', async () => {
+    // arrange: Alice creates Punks
+    // arrange: Sudo calls appPromotion.setAdminAddress(Alice)
+
+    // assert:  Random calls appPromotion.sponsorCollection(Punks.id) throws /// Random account can not set sponsoring
+    // assert:  Alice calls appPromotion.sponsorCollection(Punks.id) /// Admin account can set sponsoring
+  });
+
+  it('will set pallet address as confirmed admin for collection without sponsor', async () => {
+    // arrange: Alice creates Punks
+
+    // act:     Admin calls appPromotion.sponsorCollection(Punks.id)
+
+    // assert:  query collectionById: Punks sponsoring is confirmed by PalleteAddress
+  });
+
+  it('will set pallet address as confirmed admin for collection with unconfirmed sponsor', async () => {
+    // arrange: Alice creates Punks
+    // arrange: setCollectionSponsor(Punks.Id, Dave)
+
+    // act:     Admin calls appPromotion.sponsorCollection(Punks.id)
+
+    // assert:  query collectionById: Punks sponsoring is confirmed by PalleteAddress
+  });
+
+  it('will set pallet address as confirmed admin for collection with confirmed sponsor', async () => {
+    // arrange: Alice creates Punks
+    // arrange: setCollectionSponsor(Punks.Id, Dave)
+    // arrange: confirmSponsorship(Punks.Id, Dave)
+
+    // act:     Admin calls appPromotion.sponsorCollection(Punks.id)
+
+    // assert:  query collectionById: Punks sponsoring is confirmed by PalleteAddress
+  });
+
+  it('can be overwritten by collection owner', async () => {
+    // arrange: Alice creates Punks
+    // arrange: appPromotion.sponsorCollection(Punks.Id) 
+
+    // act:     Alice calls setCollectionSponsor(Alice)
+
+    // assert:  query collectionById: Punks sponsoring is unconfirmed by Alice
+  });
+  
+  it('will actually sponsor collection transactions in accordance with limits set by collection owner', async () => {
```

Comment:
We don't need any tests for this at the moment.

Right now, in the MVP implementation, we stick to collection limits. In later implementations, sponsoring should have its own limits.

Reply:
Split this test. 
Added a new one: "will keep collection limits set by the owner", 
and also added some assertions to "can be overwritten by collection owner"